opentelemetry-sdk/src/trace/sampler/jaeger_remote/sampling_strategy.rs

```diff
@@ -107,9 +107,10 @@ impl Inner {
                 }
             })
             .unwrap_or_else(|_err| {
-                global::handle_error(TraceError::Other(
-                    "jaeger remote sampler mutex poisoned".into(),
-                ))
+                otel_debug!(
+                    name: "JaegerRemoteSampler.MutexPoisoned",
+                    message = "Failed to update Jaeger Remote sampling strategy. The sampler's internal mutex is poisoned, indicating a panic occurred in another thread holding the lock. Sampling decisions may be using stale configuration.",
```

Comment:
Should we note that this will not recover until the remote sampling client restarts

Reply:
updated as below:
``
Unable to update Jaeger Remote sampling strategy: the sampler's internal mutex is poisoned, likely due to a panic in another thread holding the lock. This may result in using stale configuration until the remote sampling client is restarted.
``